qa/tasks/vstart_runner.py

```diff
@@ -271,6 +272,31 @@ def __init__(self):
         self.hostname = "localhost"
         self.user = getpass.getuser()
 
+    # XXX: Following two methods are vital to compatibility of tests that use
+    # teuthology.misc.sudo_write_file().
+    def sudo_write_file(self, path, data, **kwargs):
+        self.write_file(path, data, sudo=True, **kwargs)
+
+    def write_file(self, path, data, sudo=False, mode=None, owner=None,
+                   mkdir=False, append=False):
+        dd = 'sudo dd' if sudo else 'dd'
+        args = dd + ' of=' + path
+        if append:
+            args += ' conv=notrunc oflag=append'
+        if mkdir:
+            mkdirp = 'sudo mkdir -p' if sudo else 'mkdir -p'
+            dirpath = os.path.dirname(path)
+            if dirpath:
+                args = mkdirp + ' ' + dirpath + '\n' + args
+        if mode:
+            chmod = 'sudo chmod' if sudo else 'chmod'
+            args += '\n' + chmod + ' ' + mode + ' ' + path
+        if owner:
+            chown = 'sudo chown' if sudo else 'chown'
+            args += '\n' + chown + ' ' + owner + ' ' + path
+        args = shlex_split(args)
+        self.run(args=args, stdin=data)
```

Comment:
I don't know why we don't already but shouldn't `LocalRemote` inherit from `teuthology.orchestra.remote`? This would avoid all of this code duplication.

Reply:
Yes, we talked about this last time. Simply inheriting led to an error, so I'll spend some time on it and do it on a separate PR.

Reply:
This PR is meant to be a quick fix since there plenty tests in `qa/tasks/cephfs` that use `sudo_write_file()` and `write_file()`.

Reply:
Fine. Why is the method slightly different from the version in teuthology. Specifically, I'm looking at the use of `shlex_split`.

Reply:
`shlex_split` intelligently converts args from `str` to `list`. At least vstart_runner.py prefer arguments in form of list

Reply:
The teuthology `run` method's `args` option also expects a list. I'm still confused...

Reply:
Apparently, it can take string containing multiple commands separated by newlines as well. Recent update to the PR branch should handle it accordingly.